src/main/java/cpw/mods/modlauncher/TransformTargetLabel.java

```diff
@@ -103,13 +111,18 @@ public String toString() {
             this.nodeType = nodeType;
         }
 
-        public static Optional<LabelType> getTypeFor(java.lang.reflect.Type type) {
-            for (LabelType t : values()) {
-                if (t.nodeType.getName().equals(type.getTypeName())) {
-                    return Optional.of(t);
-                }
+        private static final Map<String, List<LabelType>> TYPE_LOOKUP;
+        static {
+            final Map<String, List<LabelType>> tmpTypes = new HashMap<>();
+            for (LabelType type : values()) {
+                tmpTypes.computeIfAbsent(type.nodeType.getName(), s -> new ArrayList<>()).add(type);
             }
-            return Optional.empty();
+            final Map<String, List<LabelType>> unmodifiableTypes = tmpTypes.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, pair -> Collections.unmodifiableList(pair.getValue())));
+            TYPE_LOOKUP = Collections.unmodifiableMap(unmodifiableTypes);
```

Comment:
This is in a static block to cache the resulting list, so the list and the unmodifiable wrapper don't have to be constructed on each call of `getTypeFor`